### PR TITLE
PT-1280: add help parameter and docs

### PR DESCRIPTION
**Pull request by krankenbro**
Added -? parameter which shows help for the target.
for example: vc-build Install -?

### CODE REVIEW

**Review by dimitri-ak — APPROVED** [review]

File: build/HelpProvider/Build.HelpProvider.cs [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        string result = $"{description}{Environment.NewLine}{examples}";
+        return result;
+    }
+
+    private string GetTextContent(LeafBlock leaf)
+    {
+        var inline = leaf?.Inline.FirstChild;
+        var result = new StringBuilder();
+        if (leaf == null)
+            return string.Empty;
```
Comment (Vectorfield4) [inline build/HelpProvider/Build.HelpProvider.cs:59]:
```suggestion
        if (inline is null) 
        {
            return string.Empty;
         }
```

File: build/HelpProvider/Build.HelpProvider.cs [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+            return target == GetTextContent(heading);
+        });
+        if (container == null)
+        {
+            Logger.Error($"Help is not found for the target {target}");
+            return string.Empty;
+        }
+        var descriptionBlock = container.Descendants<ParagraphBlock>().FirstOrDefault();
+        var exampleBlock = container.Descendants<FencedCodeBlock>().FirstOrDefault();
+        var description = descriptionBlock != null ? GetTextContent(descriptionBlock) : string.Empty;
```
Comment (Vectorfield4) [inline build/HelpProvider/Build.HelpProvider.cs:48]: GetTextContent already implement nullable check. No reason for checking it here.
```suggestion
        var description = GetTextContent(descriptionBlock);
```

File: build/HelpProvider/Build.HelpProvider.cs [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+                case LineBreakInline _:
+                    result.Append(Environment.NewLine);
+                    break;
+            }
+
+            if (inline.NextSibling != null)
+            {
+                inline = inline.NextSibling;
+            }
+        } while (inline != leaf.Inline.LastChild);
```
Comment (Vectorfield4) [inline build/HelpProvider/Build.HelpProvider.cs:77]:
```suggestion
            inline = inline.NextSibling;
        } while (inline != null);
```

File: build/HelpProvider/Build.HelpProvider.cs [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+        return result.ToString();
+    }
+
+    private string GetFencedText(FencedCodeBlock fencedCodeBlock)
+    {
+        if (fencedCodeBlock == null)
+            return string.Empty;
+        var lines = fencedCodeBlock.Lines.Lines.Select(l =>
```
Comment (Vectorfield4) [inline build/HelpProvider/Build.HelpProvider.cs:85]: Add an empty line after return, please.

File: build/HelpProvider/Build.HelpProvider.cs [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+        {
+            var slice = l.Slice;
+            if (EqualityComparer<StringSlice>.Default.Equals(slice, default(StringSlice)))
+                return string.Empty;
+            return slice.Text?.Substring(l.Slice.Start, l.Slice.Length) ?? string.Empty;
```
Comment (Vectorfield4) [inline build/HelpProvider/Build.HelpProvider.cs:90]: Add empty line after line 89.

File: build/HelpProvider/Build.HelpProvider.cs [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+        });
+        return string.Join(Environment.NewLine, lines);
```
Comment (Vectorfield4) [inline build/HelpProvider/Build.HelpProvider.cs:92]: Empty line.

**Review by Vectorfield4 — APPROVED** [review]
Nice work. Some styling suggestions.

**Pull request merged**